dowhy/causal_refuter.py

```diff
@@ -38,70 +38,124 @@ def get_estimator_object(new_data, identified_estimand, estimate):
+
             else:
-                # We take the side to be right as we want to find the last index that matches
-                estimate_index = np.searchsorted(simulations, estimate.value, side="right")
-                # We get the probability with respect to the left tail.
-                p_value = estimate_index / num_simulations
+                self.logger.warn("We make use of the Normal Distribution as the sample has less than 200 examples.\n \
+                Note: The underlying distribution may not be Normal. We assume that it approaches normal with the increase in sample size.")
+            
+                # Perform Normal Tests of Significance with the original estimate and the set of refutations
+                p_value = self.perform_normal_distribution_test(estimate, simulations)
+
+        elif test_type == 'bootstrap':
+            self.logger.info("Performing Bootstrap Test with {} samples\n \
+            Note: The greater the number of examples, the more accurate are the confidence estimates".format( len(simulations) ) )
+
+            # Perform Bootstrap Significance Test with the original estimate and the set of refutations
+            p_value = self.perform_bootstrap_test(estimate, simulations)
 
-            return p_value
+        elif test_type == 'normal_test':
+            self.logger.info("Performing Normal Test with {} samples\n \
+            Note: We assume that the underlying distribution is Normal.".format( len(simulations) ) )
+
+            # Perform Normal Tests of Significance with the original estimate and the set of refutations
+            p_value = self.perform_normal_distribution_test(estimate, simulations)
 
         else:
-            self.logger.warn("We make use of the Normal Distribution as the sample has less than 200 examples.\n \
-            Note: The underlying distribution may not be Normal. We assume that it approaches normal with the increase in sample size.")
-            # Get the mean for the simulations
-            mean_refute_values = np.mean(simulations)
-            # Get the standard deviation for the simulations
-            std_dev_refute_values = np.std(simulations)
-            # Get the Z Score [(val - mean)/ std_dev ]
-            z_value = (estimate.value - mean_refute_values)/ std_dev_refute_values
-            
-            if num_simulations > 30:
-                if z_value > 0.5: # Right Tail
-                    p_value = 1 - st.norm.cdf(z_value)
-                else: # Left Tail
-                    p_value = st.norm.cdf(z_value)
-            else:
-                self.logger.warn("The current evaluation has less than 30 samples. Thus, we make use of t test")
-                if z_value > 0.5: # Right Tail
-                    p_value = 1 - st.t.cdf(z_value)
-                else: # Left Tail
-                    p_value = st.t.cdf(z_value)
+            raise Exception("{} has not been implemented".format(test_type))
```

Comment:
you can use NotImplementedError to be specific :)

Reply:
cool